app/src/organisms/Devices/RobotSettings/ConnectNetwork/ConnectModal/form-state.ts

```diff
@@ -1,57 +1,62 @@
-import { useEffect } from 'react'
-import { useFormikContext, useField } from 'formik'
+import * as React from 'react'
+import {
+  useForm,
+  ControllerFieldState,
+  ControllerRenderProps,
+  FieldValues,
+} from 'react-hook-form'
 import { usePrevious } from '@opentrons/components'
 
 import type { ConnectFormValues, ConnectFormFieldProps } from '../types'
 
 export const useResetFormOnSecurityChange = (): void => {
   const {
-    values,
-    errors,
-    touched,
-    setValues,
-    setErrors,
-    setTouched,
-  } = useFormikContext<ConnectFormValues>()
+    control,
+    formState: { errors, touchedFields },
+    setValue,
+    getValues,
+    trigger,
+    clearErrors,
+  } = useForm<ConnectFormValues>()
 
-  const ssid = values.ssid
-  const ssidTouched = touched.ssid
+  const ssid = getValues('ssid')
+  const ssidTouched = touchedFields.ssid
   const ssidError = errors.ssid
-  const securityType = values.securityType
+  const securityType = getValues('securityType')
   const prevSecurityType = usePrevious(securityType)
 
-  useEffect(() => {
+  React.useEffect(() => {
```

Comment:
I think this is a ts file so we can import { useEffect } from 'react' 

```suggestion
useEffect(() => {
```